### PR TITLE
Add site header with logo

### PR DESCRIPTION
**Pull request by davidcmoulton**
Impements https://github.com/libero/pattern-library/pull/97

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,6 @@
 /translations/
 !/translations/.gitkeep
 ###< symfony/translation ###
+
+# Example asset used only directly in the pattern library
+/vendor-extra/LiberoPatternsBundle/src/Resources/public/images/custom-logo.svg
```
Comment (thewilkybarkid) [inline .gitignore]: This should be excluded from the export itself as it’s PL content.

File: vendor-extra/LiberoPatternsBundle/src/Resources/views/site-header.html.twig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+{%- set attributes = attributes|default({})|merge({class: attributes.class|default([])|merge(['site-header']) }) -%}
+
+<header {{- html.attributes(attributes) }}>
+
+    {%- set logo = logo|default({}) -%}
+    {%- with logo -%}
+
+        {%- set href = logo.href|default('') -%}
+        {%- set image = logo.image|default({})|merge({src: asset(logo.image.src|default('images/libero-logo.svg')), alt: logo.image.alt|default('Site logo')}) -%}
```
Comment (thewilkybarkid) [inline vendor-extra/LiberoPatternsBundle/src/Resources/views/site-header.html.twig]: `asset()` should only be used on pattern asset paths (ie `images/libero-logo.svg`). There’s a rewrite that hasn’t happened as a result, so this won’t work as is.

**Comment by davidcmoulton** [comment]
At the moment this is not rendering the correct markup, the implementation is missing a level:

Expected:
```
<div class="page-grid__item page-grid__start content-grid">
  <div class="content-grid__item content-grid__item--full">
    <header class="site-header">
      ...
```

Observed:
```
<div class="page-grid__item page-grid__start content-grid">
  <header class="content-grid__item site-header">
    ...
```

**Comment by davidcmoulton** [comment]
At the moment this is not rendering the correct markup, the implementation is missing a level:

Expected:
```
<div class="page-grid__item page-grid__start content-grid">
  <div class="content-grid__item content-grid__item--full">
    <header class="site-header">
      ...
```

Observed:
```
<div class="page-grid__item page-grid__start content-grid">
  <header class="content-grid__item site-header">
    ...
```

File: vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php [hunk 3]
```diff
@@ -24,7 +25,7 @@ public function __construct(TranslatorInterface $translator, UrlGeneratorInterfa
   public function onCreatePagePart(CreatePagePartEvent $event) : void
   {
 
-    $context = ['area' => null] + $event->getContext();
+    $context = ['area' => MAIN_GRID_FULL] + $event->getContext();
```
Comment (thewilkybarkid) [inline vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php]: This fixes https://github.com/libero/browser/pull/83#issuecomment-496581399. `area` in the context refers to the current grid area.

File: .gitignore (hunk 1 above)
Comment (davidcmoulton) [reply .gitignore]: Will put this in to the updated styling work for https://github.com/libero/libero/issues/139

File: vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+
```
Comment (davidcmoulton) [inline vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php]: This file needs a test.

File: vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+namespace Libero\LiberoPageBundle\EventListener;
+
+use Libero\LiberoPageBundle\Event\CreatePagePartEvent;
+use Libero\ViewsBundle\Views\ContextAwareTranslation;
+use Libero\ViewsBundle\Views\TemplateView;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use const Libero\LiberoPatternsBundle\MAIN_GRID_FULL;
+
+final class SiteHeaderListener
```
Comment (thewilkybarkid) [inline vendor-extra/LiberoPageBundle/src/EventListener/SiteHeaderListener.php]: We will need to revisit this, to probably make it dispatch an event and the details get added that way (eg to allow for custom menus), but good enough for now.

**Pull request merged**